airbyte-webapp/src/core/servicesProvider.tsx

```diff
@@ -33,6 +34,7 @@ export const ServicesProvider: React.FC<{ inject?: ServiceContainer }> = ({
       unregister: remove,
       registeredServices,
     }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Hmmm, isn't there a risk that we may forget to fix all such cases? Who will we distinguish such ignores that were intentionally added and that we just disabled for now and need to fix later?

Maybe also add some TODO comment before this, so we know that we disabled them only temporarily?

Reply:
I thought about that too. My reasoning (and please tell me if you would still prefer having TODO on everything):

All ignores I've added for `no-explicit-any` should anyway be fixed. We should not have any legit case eventually left to use `any` in (non testing) code. Thus I felt it unnecessary to add a TODO above everything, because we can still look for the ignores on `no-explicit-any` and fix them.

Regarding the `react-hooks/exhaustive-deps`. Since the webapp seems to work fine, I assume the deps list of all those hooks is actually correct, and does not need fixing, thus I believe it's just places we don't need all deps, because we have more knowledge than React (e.g. that analyticsService will never change), and thus can safely ignore them. Thus I thoughts that we actually don't need to change any in those and didn't add any TODOs to them. I'd prefer if we could teach that rule, which of our hooks are returning unchangable values, but unfortunately we can't configure this from the outside, so that eslint wouldn't complain about using some values we know that never change.

Reply:
Fixing all `no-explicit-any` sounds reasonable.

As for `react-hooks/exhaustive-deps` indeed I usually check all deps and if something is not included - it is done intentionally.

Let's then just leave it as is and fix all `no-explicit-any` later, I believe it's an awesome idea.

Reply:
I've also created https://github.com/airbytehq/airbyte/issues/10532 just now and added it to our tech-debt list so we can keep track of this.